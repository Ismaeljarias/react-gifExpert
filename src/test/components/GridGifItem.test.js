import React from "react";
import "@testing-library/jest-dom";
import { shallow } from "enzyme";

import { GridGifItem } from "../../components/GridGifItem";

describe("GridGifItem tests", () => {
  const title = "Title";
  const url = "https://localhost/";
  const wrapper = shallow(<GridGifItem title={title} url={url} />);

  test("should show <GridGifItem /> correctly", () => {
    expect(wrapper).toMatchSnapshot();
  });

  test("should should have a p tag with the Title", () => {
    const p = wrapper.find("p");
    expect(p.text().trim()).toBe(title);
  });

  test("should have img equal to url and alt from props", () => {
    const img = wrapper.find("img");
    expect(img.prop("src")).toBe(url);
    expect(img.prop("alt")).toBe(title);
  });

  test("should have animate__fadeIn", () => {
    const div = wrapper.find("div");
    const className = div.prop("className");

    expect(className.includes("animate__fadeIn")).toBe(true);
  });
});
