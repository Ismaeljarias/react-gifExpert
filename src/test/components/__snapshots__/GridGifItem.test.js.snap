// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GridGifItem tests should show <GridGifItem /> correctly 1`] = `
<div
  className="card animate__animated animate__fadeIn"
>
  <img
    alt="Title"
    src="https://localhost/"
  />
  <p>
    Title
  </p>
</div>
`;
